File: 2428-equal-row-and-column-pairs/equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        int ans=0;
        map<vector<int>,int>map;
        int n=grid.size();

        for(int i=0;i<n;i++){
            map[grid[i]]++;
        }
        
        for(int i=0;i<n;i++){
            vector<int>temp;
            for(int j=0;j<n;j++){
                temp.push_back(grid[j][i]);
            }
            if(map[temp]>0){
                ans+=map[temp];
            }
        }
        return ans;
    }
};

File: 2428-equal-row-and-column-pairs/equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: list[list[int]]) -> int:
        n=len(grid)
        col_grid=[]
        for i in range(n):
            row=[]
            col=[]
            for j in range(n):
                col.append(grid[j][i])
            col_grid.append(col)
        ans=0
        idx=[]
        for i in range(n):
            if grid[i] in col_grid:
                for ele in col_grid:
                    if ele ==grid[i]:
                        ans+=1
        return ans
        

        